app/workers/dormant_user_mailer_worker.rb

```diff
@@ -0,0 +1,12 @@
+class DormantUserMailerWorker
+  include Sidekiq::Worker
+
+  attr_reader :user
+
+  def perform(user_id)
+    @user = User.find(user_id)
```

Comment:
Does the mailer need the instance vars here?  Looks like it'll assign it's own instance vars, remove them if not. Also you can use the safe navigation operator here if you like:
```
@user = User.find(user_id)
if @user&.receives_email?
  DormantUserMailer.email_dormant_user(@user).deliver
end 
```
Also nice work on the check that this user can receive email, top work 👍

Reply:
I'm not sure what cam meant, but I think he meant that the instance variables here could just be normal variables, at least in the worker. Cam's suggestion of the safe navigation operator was a nice idea, except `User.find` will raise an exception, not return nil. You could rescue `ActiveRecord::RecordNotFound` and just do nothing in that case, but with User we never ever remove user records from the database, so if you can't find them, it really is something strange going on and we probably want the default behaviour (which sends us an email about the exception, and retries automatically).

```
def perform(user_id)
  user = User.find(user_id)
  
  if user.receives_email?
    DormantUserMailer.email_dormant_user(user).deliver
  end
end
```